hummingbot/connector/derivative/coinflex_perpetual/coinflex_perpetual_web_utils.py

```diff
@@ -0,0 +1,314 @@
+    if only_hostname:
+        return CONSTANTS.REST_URL.format(subdomain_prefix)
+    return "https://" + CONSTANTS.REST_URL.format(subdomain_prefix) + f"/{local_endpoint_api_version}{endpoint}"
+
+
+def private_rest_url(path_url: str = "",
+                     domain: str = CONSTANTS.DEFAULT_DOMAIN,
+                     only_hostname: bool = False,
+                     domain_api_version: str = None,
+                     endpoint_api_version: str = None) -> str:
+    """
+    Creates a full URL for provided private REST endpoint
+    :param path_url: a private REST endpoint
+    :param host: the CoinFLEX host to connect to
+    :return: the full URL to the endpoint
+    """
+    local_domain_api_version = domain_api_version or CONSTANTS.PRIVATE_API_VERSION
+    local_endpoint_api_version = endpoint_api_version or CONSTANTS.PRIVATE_API_VERSION
+    subdomain_prefix = f"{local_domain_api_version}stg" if domain == "coinflex_perpetual_testnet" else local_domain_api_version
+    endpoint = "" if not len(path_url) else f"/{path_url}"
+    if only_hostname:
+        return CONSTANTS.REST_URL.format(subdomain_prefix)
+    return "https://" + CONSTANTS.REST_URL.format(subdomain_prefix) + f"/{local_endpoint_api_version}{endpoint}"
+
+
+def private_rest_auth_path(path_url: str,
+                           domain: str = CONSTANTS.DEFAULT_DOMAIN,
+                           endpoint_api_version: str = None) -> str:
+    """
+    Creates an auth URL path for provided private REST endpoint
+    :param path_url: a private REST endpoint
+    :param host: the CoinFLEX host to connect to
+    :return: the auth URL path for the endpoint
+    """
+    local_endpoint_api_version = endpoint_api_version or CONSTANTS.PRIVATE_API_VERSION
+    return f"/{local_endpoint_api_version}/{path_url}"
+
+
+def websocket_url(domain: str = CONSTANTS.DEFAULT_DOMAIN,
+                  domain_api_version: str = None,
+                  endpoint_api_version: str = None) -> str:
+    """
+    Creates a full URL for provided public REST endpoint
+    :param host: the CoinFLEX host to connect to
+    :return: the full URL to the endpoint
+    """
+    local_domain_api_version = domain_api_version or CONSTANTS.PUBLIC_API_VERSION
+    local_endpoint_api_version = endpoint_api_version or CONSTANTS.PUBLIC_API_VERSION
+    subdomain_prefix = f"{local_domain_api_version}stg" if domain == "coinflex_perpetual_testnet" else local_domain_api_version
+    return CONSTANTS.WSS_URL.format(subdomain_prefix, local_endpoint_api_version)
+
+
+def build_api_factory(auth: Optional[AuthBase] = None) -> WebAssistantsFactory:
+    api_factory = WebAssistantsFactory(auth=auth, rest_pre_processors=[CoinflexPerpetualRESTPreProcessor()])
+    return api_factory
+
+
+def create_throttler() -> AsyncThrottler:
+    return AsyncThrottler(CONSTANTS.RATE_LIMITS)
+
+
+@dataclass
+class CoinflexPerpetualRESTRequest(EndpointRESTRequest):
+    """
+    CoinFLEX version of the `RESTRequest` class.
+    """
```

Comment:
It does not seem to be required to have a special RESTRequest implementation for the connector. The logic included here could be part of the Auth class and the generic RESTRequest could be used instead.

Reply:
Not doing that sorry, this is far cleaner in my opinion

Reply:
Ok, my comment was more a nice to have than a change request. I will approve this change.